src/lib/ext-activitylog.js

```diff
@@ -1,30 +1,47 @@
-import { save } from '../lib/save-load.js';
+import { save } from './save-load.js';
 
-export default class ActivityLog {
+class Model {
   constructor() {
     this.logs = [];
+  }
+
+  addNewLogs(logs) {
+    this.logs.push(...logs);
+  }
+}
 
+class View {
+  constructor() {
+    this.logTable = document.querySelector('log-view');
     this.saveLogBtn = document.getElementById('saveLogBtn');
-    this.showLogDetailWrapper = document.querySelector(
-      '.show-log-detail-wrapper'
-    );
-    this.showLogDetails = document.getElementById('showLogDetails');
-    this.logTableWrapper = document.querySelector('.log-table');
-    this.tableBody = document.querySelector('table tbody');
     this.notice = document.querySelector('.notice');
-    this.closeBtn = document.querySelector('.close');
   }
 
-  async init() {
-    const existingLogs = await this.getExistingLogs();
+  addTableRows(logs) {
+    this.logTable.addNewRows(logs);
+  }
 
-    if (existingLogs.length) {
-      this.addNewLogs(existingLogs);
+  setError(errorMessage) {
+    if (errorMessage) {
+      this.notice.textContent = errorMessage;
+      this.notice.classList.add('failure');
+    } else {
+      this.notice.textContent = '';
+      this.notice.classList.remove('failure');
     }
+  }
+}
+
+class Controller {
+  constructor(model, view) {
+    this.model = model;
+    this.view = view;
+
+    this.init();
+  }
 
-    this.saveLogBtn.addEventListener('click', this);
-    this.closeBtn.addEventListener('click', this);
-    this.tableBody.addEventListener('click', this);
+  async init() {
+    this.view.saveLogBtn.addEventListener('click', this);
```

Comment:
To decouple the view's DOM from this high-level View class, subscribe for a custom event on the view. In the view, you can use the `CustomEvent` to create a new event and fire it. See https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Creating_and_triggering_events

Reply:
done :-)